adapters/httpapi_winhttp.c

```diff
@@ -820,6 +977,61 @@ HTTPAPI_RESULT HTTPAPI_CloneOption(const char* optionName, const void* value, co
                 result = HTTPAPI_OK;
             }
         }
+        else if (strcmp(OPTION_HTTP_PROXY, optionName) == 0)
+        {
+            *savedValue = malloc(sizeof(HTTP_PROXY_OPTIONS));
+            if (*savedValue == NULL)
+            {
+                LogError("failure in malloc");
+                result = HTTPAPI_ERROR;
+            }
+            else
+            {
+                HTTP_PROXY_OPTIONS *savedOption = (HTTP_PROXY_OPTIONS*) *savedValue;
+                HTTP_PROXY_OPTIONS *option = (HTTP_PROXY_OPTIONS*) value;
+                if (mallocAndStrcpy_s((char**)&(savedOption->host_address),
+                                      (const char*)option->host_address) != 0)
+                {
+                  LogError("unable to clone the proxy host adress content");
+                  free((void*)*savedValue);
+                  *savedValue = NULL;
+                  result = HTTPAPI_ERROR;
+                }
+                else
+                {
+                    savedOption->port = option->port;
+                    if (option->username != NULL && mallocAndStrcpy_s((char**)&(savedOption->username),
```

Comment:
Oops, found a bug here. Sorry.

At lines 1003 and 1015 savedOption->username and savedOption->password are only set if option->username or option->password are not NULL, respectively.
So if any is NULL the respective value in savedOption is not set.
However, savedOption values as never NULLed initially, so they will have invalid pointer addresses (gargabe) set in this case, which later can cause a memory access violation.

Could you please add this call between lines 991 and 992?

memset(savedValue, 0, sizeof(HTTP_PROXY_OPTIONS));

Reply:
Oops, missed that too! Fixed.

Reply:
Just an update, I had a typo on the command above.
I fixed it myself to be:
memset((void*)*savedValue, 0, sizeof(HTTP_PROXY_OPTIONS));